src/crystal/system/unix/process.cr

```diff
@@ -0,0 +1,262 @@
+    argv << Pointer(UInt8).null
+
+    LibC.execvp(command, argv)
+
+    error_message = String.build do |io|
+      io << "execvp ("
+      command.inspect_unquoted(io)
+      args.try &.each do |arg|
+        io << ' '
+        arg.inspect(io)
+      end
+      io << ")"
+    end
+    raise Errno.new(error_message)
+  end
+
+  protected def self.prepare_shell_system(command, args)
+    command = %(#{command} "${@}") unless command.includes?(' ')
+    shell_args = ["-c", command, "--"]
+
+    if args
+      unless command.includes?(%("${@}"))
+        raise ArgumentError.new(%(can't specify arguments in both, command and args without including "${@}" into your command))
+      end
+
+      {% if flag?(:freebsd) %}
+        shell_args << ""
+      {% end %}
+
+      shell_args.concat(args)
+    end
+
+    command = "/bin/sh"
+    {command, shell_args}
+  end
+
+  # :nodoc:
+  protected def self.fork_internal(*, will_exec : Bool) : Int64?
+    newmask = uninitialized LibC::SigsetT
+    oldmask = uninitialized LibC::SigsetT
+
+    LibC.sigfillset(pointerof(newmask))
+    ret = LibC.pthread_sigmask(LibC::SIG_SETMASK, pointerof(newmask), pointerof(oldmask))
+    raise Errno.new("Failed to disable signals") unless ret == 0
+
+    case pid = LibC.fork
+    when 0
+      # child:
+      pid = nil
+      if will_exec
+        # reset signal handlers, then sigmask (inherited on exec):
+        Crystal::Signal.after_fork_before_exec
+        LibC.sigemptyset(pointerof(newmask))
+        LibC.pthread_sigmask(LibC::SIG_SETMASK, pointerof(newmask), nil)
+      else
+        {% unless flag?(:preview_mt) %}
+          ::Process.after_fork_child_callbacks.each(&.call)
+        {% end %}
+        LibC.pthread_sigmask(LibC::SIG_SETMASK, pointerof(oldmask), nil)
+      end
+    when -1
+      # error:
+      errno = Errno.value
+      LibC.pthread_sigmask(LibC::SIG_SETMASK, pointerof(oldmask), nil)
+      raise Errno.new("fork", errno)
+    else
+      # parent:
+      LibC.pthread_sigmask(LibC::SIG_SETMASK, pointerof(oldmask), nil)
+    end
+
+    pid.try &.to_i64
+  end
+
+  protected def close_system
+  end
+
+  protected def self.chroot_system(path : String) : Nil
+    path.check_no_null_byte
+    if LibC.chroot(path) != 0
+      raise Errno.new("Failed to chroot")
+    end
+
+    if LibC.chdir("/") != 0
+      errno = Errno.new("chdir after chroot failed")
+      errno.callstack = CallStack.new
+      errno.inspect_with_backtrace(STDERR)
+      abort("Unresolvable state, exiting...")
+    end
+  end
+
+  protected def self.times_system : Tms
+    LibC.getrusage(LibC::RUSAGE_SELF, out usage)
+    LibC.getrusage(LibC::RUSAGE_CHILDREN, out child)
+
+    Tms.new(
+      usage.ru_utime.tv_sec.to_f64 + usage.ru_utime.tv_usec.to_f64 / 1e6,
+      usage.ru_stime.tv_sec.to_f64 + usage.ru_stime.tv_usec.to_f64 / 1e6,
+      child.ru_utime.tv_sec.to_f64 + child.ru_utime.tv_usec.to_f64 / 1e6,
+      child.ru_stime.tv_sec.to_f64 + child.ru_stime.tv_usec.to_f64 / 1e6,
```

Comment:
The `#to_f64` are not needed anymore since https://github.com/crystal-lang/crystal/pull/8120

Reply:
Good catch. But I'd prefer to put such changes in a separate PR.